pubsub/iterator.go

```diff
@@ -261,9 +256,6 @@ func (it *messageIterator) receive(maxToPull int32) ([]*Message, error) {
 	maxExt := time.Now().Add(it.po.maxExtension)
 	ackIDs := map[string]*AckResult{}
 	it.mu.Lock()
-	it.eoMu.RLock()
-	enableExactlyOnceDelivery := it.enableExactlyOnceDelivery
```

Comment:
This is the only bit that bears scrutiny.  Will not setting this affect any other behaviors?

Reply:
This was only for reading the field from the iterator object into a local variable. It was used for this function only.